search/search_repository.go

```diff
@@ -329,12 +331,14 @@ func parseOptions(queryMap map[string]interface{}) *QueryOptions {
 					options.ParentExists = v.(bool)
 				case OptTreeViewKey:
 					options.TreeView = v.(bool)
+				case OptChildIterations:
+					options.ChildIterations = v.(bool)
 				}
 			}
 			return &options
 		}
 	}
-	return nil
+	return &QueryOptions{}
```

Comment:
@baijum why return an empty struct? It's a pointer and `nil` makes more sense to me.

Reply:
With `nil`, wherever the attribute is accessed, a check is required to ensure the value is not nil. Whereas with struct with default value makes easy to program. I have also changed one default value also.